modules/tinymce/src/themes/silver/main/ts/ui/core/color/Options.ts

```diff
@@ -135,20 +152,10 @@ const colorColsOption = (editor: Editor, id: string): number => {
 };
 
 const getColorCols = (editor: Editor, id: string): number => {
+  // Sensible default based on the amount of colors in 'color_map'
+  const defaultCols = Math.ceil(Math.sqrt(option('color_map')(editor).length));
```

Comment:
I suspect a lot of this can be simplified by switching the cols options to use a validator function rather than simply a number processor. Once that's done you can assume the three cols options are never undefined or less than 1.

I've been trying to hint at that for a while but I'm clearly not being obvious enough 😂

Reply:
Moved number validation to a processor, this function now just retrieves the rounded value.